core/shared/src/main/scala/org/http4s/internal/UriCoding.scala

```diff
@@ -24,7 +24,7 @@ import java.nio.charset.{Charset => JCharset}
 /* Exists to work around circular dependencies */
 private[http4s] object UriCoding {
   val Unreserved: CharPredicate = AlphaNum ++ "-_.~"
-  val QueryNoEncode: CharPredicate = Unreserved ++ "?/"
+  val QueryNoEncode: CharPredicate = Unreserved ++ "!$&'()*+,;=:/?@"
```

Comment:
JFTR, the current implementation is the same as defined in [Wikipedia](https://en.wikipedia.org/wiki/Query_string#URL_encoding) 🤔

Reply:
Yes, I know it. However, if @armanbilge says the same thing as you that we should stay with the one that Wikipedia "approves". Then I will redo my changes and will modify my solution.

Reply:
@danicheg Updated the PR. Probably the checks will fail on two test cases:
1. The first failing test is this one:
https://github.com/http4s/http4s/blob/da8c65b4702f0fe7b073e31b384416d88ec39078/tests/shared/src/test/scala/org/http4s/UriSuite.scala#L1423
This should not be accepted as valid uri during parsing: https://en.wikipedia.org/wiki/Query_string#URL_encoding:~:text=The%20series%20of%20pairs%20is%20separated

2. The second failing test is:
https://github.com/http4s/http4s/blob/da8c65b4702f0fe7b073e31b384416d88ec39078/tests/shared/src/test/scala/org/http4s/UriSuite.scala#L559
This should not be accepted anymore, because brackets should be encoded in the query parameters